### PR TITLE
Bump stack.yaml dependencies to support cardano-node 1.8

### PR DESCRIPTION
**Pull request by Anviking**
Relates to #1447. Not sure to what extent / if this is strictly required for it, so I opted to create a separate PR.

# Overview

- [x] Adapted to: iohk-monitoring-framework's `Trace` is now a Tracer (**LoggerName**, LogObject a)
- [x] Adapt to: ourobouros-network has changed a bunch of things (https://github.com/input-output-hk/cardano-db-sync/pull/20/files looks helpful for guidance)

# Comments

### CODE REVIEW

**Comment by Anviking** [comment]
@KtorZ I'm now able to compile cardano-wallet-byron, but it appears that it is silently not syncing. Any idea?

_maybe I just had the wrong socket path_

File: lib/byron/src/Cardano/Wallet/Byron.hs [hunk 1]
```diff
@@ -181,7 +181,7 @@ serveWallet
     -- ^ Which host to bind.
     -> Listen
     -- ^ HTTP API Server port.
-    -> AddrInfo
+    -> FilePath
```
Comment (KtorZ) [inline lib/byron/src/Cardano/Wallet/Byron.hs:184]: why reverting this to a 'FilePath' :thinking: ? `AddrInfo` is actually clearer about what this parameter is. A `FilePath` can be anything..

File: lib/byron/src/Cardano/Wallet/Byron/Network.hs [hunk 2]
```diff
@@ -336,23 +344,12 @@ send queue cmd = do
...
-        --  - Tx submission
-    m
-        -- Underlying monad we run in
     ByteString
         -- Concrete representation for bytes string
+    m
+        -- Underlying monad we run in
     Void
-        -- Return type of a network client. Void indicates that the client
-        -- never exits.
```
Comment (Anviking) [inline lib/byron/src/Cardano/Wallet/Byron/Network.hs:352]: Comments got lost (wasn't sure if they still were valid at some intermediary point)

File: lib/byron/src/Cardano/Wallet/Byron/Compatibility.hs [hunk 3]
```diff
@@ -310,7 +319,7 @@ fromBlockNo (BlockNo h) =
     Quantity (fromIntegral h)
 
 fromTip :: W.Hash "Genesis" -> W.EpochLength -> Tip ByronBlock -> W.BlockHeader
-fromTip genesisHash epLength tip = case getPoint (tipPoint tip) of
+fromTip genesisHash epLength tip = case getPoint (fst $ toLegacyTip tip) of
```
Comment (KtorZ) [inline lib/byron/src/Cardano/Wallet/Byron/Compatibility.hs]: `fst $ toLegacyTip tip` --> `getTipPoint tip`

File: lib/byron/src/Cardano/Wallet/Byron/Compatibility.hs [hunk 4]
```diff
@@ -319,7 +328,7 @@ fromTip genesisHash epLength tip = case getPoint (tipPoint tip) of
         }
     At blk -> W.BlockHeader
         { slotId = fromSlotNo epLength $ Point.blockPointSlot blk
-        , blockHeight = fromBlockNo $ tipBlockNo tip
+        , blockHeight = fromBlockNo $ snd $ toLegacyTip tip
```
Comment (KtorZ) [inline lib/byron/src/Cardano/Wallet/Byron/Compatibility.hs]: `snd $ toLegacyTip tip` --> `getLegacyTipBlockNo tip`

File: lib/byron/src/Cardano/Wallet/Byron/Network.hs [hunk 5]
```diff
@@ -367,29 +364,33 @@ mkNetworkClient
...
     -> (NodeToClientVersionData, CodecCBORTerm Text NodeToClientVersionData)
-    -> AddrInfo
+    -> FilePath
     -> IO ()
-connectClient client (vData, vCodec) addr = do
+connectClient client (vData, vCodec) addr = withIOManager $ \iocp -> do
     let vDict = DictVersion vCodec
     let versions = simpleSingletonVersions NodeToClientV_1 vData vDict client
     let tracers = NetworkConnectTracers nullTracer nullTracer
+    let socket = localSnocket iocp addr
```
Comment (KtorZ) [inline lib/byron/src/Cardano/Wallet/Byron/Network.hs:395]: Okay. So this is why we went from `AddrInfo` -> `FilePath`

File: lib/byron/src/Cardano/Wallet/Byron/Network.hs (hunk 2 above)
Comment (KtorZ) [reply lib/byron/src/Cardano/Wallet/Byron/Network.hs:352]: Still valid as far as I am concerned.

File: lib/byron/src/Cardano/Wallet/Byron/Compatibility.hs [hunk 6]
```diff
@@ -181,8 +182,16 @@ emptyGenesis bp = W.Block
...
+
 genesisTip :: Tip ByronBlock
-genesisTip = Tip genesisPoint genesisBlockNo
+genesisTip = legacyTip genesisPoint genesisBlockNo
+  where
+    -- NOTE: ourobouros-network states that:
+    --
+    -- There /is/ no block number if we are at genesis
+    -- ('genesisBlockNo' is the block number of the first block on the chain).
+    -- Usage of this function should be phased out.
```
Comment (KtorZ) [inline lib/byron/src/Cardano/Wallet/Byron/Compatibility.hs:194]: :+1:

**Pull request merged**